EpsilonLights/Tests/tests/LightsTest/LightsTest.c

```diff
@@ -0,0 +1,373 @@
+#include "unity.h"
+#include "Mockstm32f4xx_hal_gpio.h"
+#include "Mockstm32f4xx_hal_can.h"
+#include "Mockcmsis_os.h"
+#include "Lights.h"
+#include "LightsTest.h"
+
+LightsInfo lightsInfo = {0};
+RegenBrakeInfo regenBrakeInfo = {0};
+
+uint8_t lightsInputs;
+uint8_t auxBmsInputs[2];
+uint8_t batteryErrors[5];
+uint8_t driversInputs[4];
+SigLightsHandle sigLightsHandle;
+
+osMutexId* canHandleMutex;
+
+void runLightsTests()
+{
+    RUN_TEST(test_updateLights1_allLightsOff);
+    RUN_TEST(test_updateLights1_headLightsLowBreakLightsOnHazardsOnBmsStrobeLightOn);
+    RUN_TEST(test_updateLights1_headLightsHighHeadLightsLowBreakLightsOnHazardsOnBmsStrobeLightOn);
+    RUN_TEST(test_blinkSignalLights_sigLightsLeftOFFSigLightsRightOFF);
+    RUN_TEST(test_blinkSignalLights_sigLightsLeftONSigLightsRightONPrevSigStateZero);
+    RUN_TEST(test_blinkSignalLights_prevSigState1SigLightsLeftONSigLightsRightON);
+    RUN_TEST(test_updateStrobeLights1_ESTROBEONUpdateBlinkerTimer);
+    RUN_TEST(test_updateStrobeLights1_ESTROBEOFFResetBlinkerTimer);
+    RUN_TEST(test_updateStrobeLights1_ESTROBEOFFUpdateBlinkerTimer);
+    RUN_TEST(test_reportLightsToCan);
+    RUN_TEST(test_sendHeartbeat);
+    RUN_TEST(test_HAL_CAN_RxCpltCall_backLightsInputStdId);
+    RUN_TEST(test_HAL_CAN_RxCpltCallback_batteryStatErrorsStdId);
+    RUN_TEST(test_HAL_CAN_RxCpltCallback_driversInputsStdId);
+    RUN_TEST(test_HAL_CAN_RxCpltCallback_auxBmsInputStdId);
+    RUN_TEST(test_HAL_CAN_RxCpltCallback_batteryStatErrorsStdIdANDToggleRedPin);
+}
+
+/*Test updateLights(). Headlights, brakes, hazards and bms strobe are
+off. Checks if lightsInfo.leftSignal matches sigLightsHandle.left and 
+lightsInto.rightSignal matches sigLightsHandle.right  */
+void test_updateLights1_allLightsOff()
```

Comment:
Please update your names for the test to reflect the name changes you've made in Lights.c.